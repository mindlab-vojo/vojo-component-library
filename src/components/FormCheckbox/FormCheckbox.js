import React,  { useState, useEffect } from 'react'
import PropTypes from 'prop-types'

import ThemeConsumer from '../../style/ThemeManager/ThemeProvider'
import * as FormCheckbox from './FormCheckbox.style'
import IconManager from '../IconManager'
import Typography from '../Typography'

const FormCheckboxComponent = ({
  description,
  errorMessage,
  hasError,
  hasToolTip,
  onClickOption,
  onClickToolTip,
  options,
  preSelectedOptions,
  title,
  tooltipData,
}) => {
  const createPreselectedOptions = () => {
    const newOptionsToState = []

    for (let i = 0; i < preSelectedOptions.length; i++) {
      const preSelectedOption = preSelectedOptions[i];
      newOptionsToState.push(
        options.find(option => option.id === preSelectedOption)
      )
    }
    
    return newOptionsToState
  }

  const defaultOption = preSelectedOptions ? createPreselectedOptions() : []
  const [selectedOptions, setSelectedOptions] = useState(defaultOption)

  const handleClickOption = (option) => {
    hasOptionOnSelectedOptions(option)
      ? removeOption(option)
      : addOption(option)
  }

  const hasOptionOnSelectedOptions = (option) => {
    const searchForOptionResult = selectedOptions.find(
      selectedOption => selectedOption.id === option.id
    )
    return Boolean(searchForOptionResult)
  }

  const addOption = (option) => {
    const newSelectedOptions = [...selectedOptions]
    newSelectedOptions.push(option)
    setSelectedOptions(newSelectedOptions)
    onClickOption && onClickOption(newSelectedOptions)
  }

  const removeOption = (option) => {
    const newSelectedOptions = selectedOptions.filter(
      selectedOption => selectedOption.id !== option.id
    )
    setSelectedOptions(newSelectedOptions)
    onClickOption && onClickOption(newSelectedOptions)
  }

  const handleClickToolTip = () => {
    onClickToolTip && onClickToolTip(tooltipData)
  }

  const renderOptions = (theme) => {
    return options.map(option => {
      const isSelected = hasOptionOnSelectedOptions(option)
      return (
        <FormCheckbox.Option
          key={option.id}>
          <FormCheckbox.HiddenCheckbox
            checked={isSelected}
            type="checkbox"
            value={option.value}
            id={option.id}
            onChange={() => null}
          />
          <FormCheckbox.Label
            htmlFor={option.id}
            onClick={() => handleClickOption(option)}>
            <FormCheckbox.CheckboxIcon>
              <IconManager 
                height="20px"
                width="20px"
                icon={
                  isSelected ?
                    "CheckboxChecked" : 
                    "CheckboxUnchecked"
                }
                fill={theme.colors.blackColor}/>
            </FormCheckbox.CheckboxIcon>
            <Typography
              color={theme.colors.blackColor}
              fontSize="14px">
              {option.title}
            </Typography>
          </FormCheckbox.Label>
        </FormCheckbox.Option>
      )
    })
  }

  const renderComponent = (theme) => {
    return (
      <FormCheckbox.Wrapper>
        <FormCheckbox.Header>
          <FormCheckbox.TitleWrapper>
            <FormCheckbox.Title>
              <Typography
                color={hasError ? theme.colors.redColor : theme.colors.blackColor}
                fontSize="16px"
                fontWeight="500">
                {title}
              </Typography>
            </FormCheckbox.Title>
            {
              description && (
                <FormCheckbox.Description>
                  <Typography
                    color={hasError ? theme.colors.redColor : theme.colors.blackColor}
                    fontSize="12px">
                    {description}
                  </Typography>
                </FormCheckbox.Description>
              )
            }
            {
              hasError && (
                <FormCheckbox.Description>
                  <Typography
                    color={theme.colors.redColor}
                    fontSize="12px">
                    {errorMessage}
                  </Typography>
                </FormCheckbox.Description>
              )
            }
          </FormCheckbox.TitleWrapper>
          {
            hasToolTip &&(
              <FormCheckbox.Tooltip
                onClick={handleClickToolTip}>
                <IconManager 
                  height="20px"
                  width="20px"
                  icon="HelpCircle"
                  fill={theme.colors.blackColor}/>
              </FormCheckbox.Tooltip>
            )
          }
        </FormCheckbox.Header>
        <FormCheckbox.OptionsWrapper>
          <FormCheckbox.OptionsWrapper>
            {
              renderOptions(theme)
            }
          </FormCheckbox.OptionsWrapper>
        </FormCheckbox.OptionsWrapper>
      </FormCheckbox.Wrapper>
    )
  }

  return (
    <ThemeConsumer>
      {({ theme }) => renderComponent(theme)}
    </ThemeConsumer>
  )
}

FormCheckboxComponent.propTypes = {
  description: PropTypes.string,
  errorMessage: PropTypes.string,
  hasError: PropTypes.bool,
  hasToolTip: PropTypes.bool,
  onClickOption: PropTypes.func,
  onClickToolTip: PropTypes.func,
  options: PropTypes.array,
  preSelectedOptions: PropTypes.array,
  title: PropTypes.string,
  tooltipData: PropTypes.object,
}

export default FormCheckboxComponent
