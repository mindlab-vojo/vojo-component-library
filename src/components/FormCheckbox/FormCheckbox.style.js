import styled from 'styled-components'

export const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  width: 100%;
`

export const Header = styled.div`
  display: flex;
  justify-content: space-between;
  width: 100%;
`

export const TitleWrapper = styled.div`
  display: flex;
  flex-direction: column;
`

export const Title = styled.div`
  display: flex;
`

export const Tooltip = styled.div`
  cursor: pointer;
  display: flex;
`

export const Description = styled.div`
  display: flex;
`

export const OptionsWrapper = styled.div`
  display: flex;
  flex-direction: column;
`

export const Option = styled.div`
  display: flex;
  margin: 5px 0;

  &:first-of-type { 
    margin-top: 12px;
  }
`

export const HiddenRadio = styled.input`
  display: none;
`

export const RadioIcon = styled.div`
  margin-right: 5px;
`

export const Label = styled.label`
  cursor: pointer;
  display: flex;
`