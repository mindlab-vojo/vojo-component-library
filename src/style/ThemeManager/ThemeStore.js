import { vojo } from './../themes'

class ThemeStore {

  constructor(theme) {
    this.defaultTheme = vojo
    this.theme = theme ? theme : this.defaultTheme
  }

  configure = ({ theme }) =>  {
    if (theme) this.configureTheme(theme)
    else this.configureTheme(this.defaultTheme)
  }

  configureTheme = (theme) => {
    this.theme = theme
  }

  modify = (modifiedProps) => {
    const currentTheme = this.theme
    return { currentTheme, modifiedProps }
  }

  getTheme = () => this.theme

}

export default ThemeStore