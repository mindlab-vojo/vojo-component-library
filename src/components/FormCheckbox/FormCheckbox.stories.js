import React from 'react';
import { withKnobs, text, array, boolean, object } from '@storybook/addon-knobs';
import { withA11y } from '@storybook/addon-a11y';
import { action } from '@storybook/addon-actions'

import FormCheckbox from '.'; 
import { ThemeProvider } from "../../../src";
import categories from '../../../.storybook/categories';

export const formCheckbox = () => {
  const optionsList = [
    {
      id: '1',
      title: 'Opção 1',
      value: 'leonardo',
    },
    {
      id: '2',
      title: 'Opção 2',
      value: 'renata',
    },
    {
      id: '3',
      title: 'Opção 3',
      value: 'robson',
    },
    {
      id: '4',
      title: 'Opção 4',
      value: 'ana-carolina',
    },
  ]

  const tooltipDataObject = {
    title: "Título do Modal",
    description: "Descrição que aparecerá no modal, normalmente descreverá detalhes dos requisistos deste campo."
  }

  const preSelectedOptions = array('preSelectedOptions', ['1', '3'])
  const errorMessage = text('errorMessage', 'Este campo possui algum erro...')
  const hasError = boolean('hasError', false)
  const hasToolTip = boolean('hasToolTip', true)
  const tooltipData = object('tooltipData', tooltipDataObject)
  const onClickOption = action('onClickOption')
  const onClickToolTip = action('onClickToolTip')
  const options = object('options', optionsList)
  const description = text('description', 'Breve descrição da pergunta')
  const title = text('title', 'Título da Pergunta')
  const hasAllOptions = boolean('hasAllOptions', true)

  return (
    <FormCheckbox
      description={description}
      errorMessage={errorMessage}
      hasAllOptions={hasAllOptions}
      hasError={hasError}
      hasToolTip={hasToolTip}
      onClickOption={onClickOption}
      onClickToolTip={onClickToolTip}
      options={options}
      preSelectedOptions={preSelectedOptions}
      tooltipData={tooltipData}
      title={title}
    />
  )
}

export default {
  title: categories.Radios,
  component: formCheckbox,
  decorators: [
    withKnobs, 
    withA11y, 
    (story) => <ThemeProvider>{ story() }</ThemeProvider>,
    (story) => <div style={{ display: "flex" }}>{ story() }</div>,
  ],
}

formCheckbox.story = {
  name: 'FormCheckbox',
};
  