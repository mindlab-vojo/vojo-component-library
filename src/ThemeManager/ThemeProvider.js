import React, { Component, createContext } from 'react'
import PropTypes from 'prop-types'

import ThemeStore from './ThemeStore';

// Provider and Consumer instances
const { Provider, Consumer } = createContext();

class ThemeProvider extends Component {
  static propTypes = {
    children: PropTypes.node,
    theme: PropTypes.string
  }

  state = {
    themeConfig: new ThemeStore()
  }

  componentDidMount() {
    if (this.props.theme) this.setTheme()
  }

  componentDidUpdate(prevProps) {
    if (this.props.theme !== prevProps.theme) this.setTheme()
  }

  setTheme = () => {
    const { theme } = this.props
    const { themeConfig } = this.state
    const updatedTheme = themeConfig.configure({ theme })
    this.setState({ theme: updatedTheme })
  }

  render() {
    return (
      <Provider value={{ theme: this.state.themeConfig.theme }}>
        {this.props.children}
      </Provider>
    )
  }
}

export { ThemeProvider }

export default Consumer