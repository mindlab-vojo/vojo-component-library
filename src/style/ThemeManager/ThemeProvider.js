import React, { Component, createContext } from 'react'
import PropTypes from 'prop-types'

import ThemeStore from './ThemeStore';

// Provider and Consumer instances
const { Provider, Consumer } = createContext();

class ThemeProvider extends Component {
  static propTypes = {
    children: PropTypes.node,
    theme: PropTypes.string
  }

  state = {
    themeConfig: new ThemeStore()
  }

  componentDidMount() {
    if (this.props.theme) this.setTheme()
  }

  componentDidUpdate(prevProps) {
    if (this.props.theme !== prevProps.theme) this.setTheme()
  }

  setTheme = () => {
    const { theme } = this.props
    const { themeConfig } = this.state
    const updatedTheme = themeConfig.configure({ theme })
    this.setState({ theme: updatedTheme })
  }

  getThemeFont = () => {
    const { themeConfig } = this.state
    return themeConfig.font
  }

  render() {

    const GlobalFont = this.getThemeFont()

    return (
      <Provider value={{ theme: this.state.themeConfig.theme }}>
        <GlobalFont/>
        {this.props.children}
      </Provider>
    )
  }
}

export { ThemeProvider }

export default Consumer