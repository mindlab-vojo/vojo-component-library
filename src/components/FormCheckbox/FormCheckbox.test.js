import React from 'react';
import { render } from '@testing-library/react';

import FormCheckbox from './';
import { ThemeProvider } from './../../style/ThemeManager/ThemeProvider';

test('Test FormCheckbox component render', () => {
  const fakeFn = jest.fn()
  const component = render(
    <ThemeProvider>
      <FormCheckbox 
        description="Este é um exemplo de descrição"
        errorMessage="Este é um erro"
        hasError={true}
        hasToolTip={true}
        onClickOption={fakeFn}
        onClickToolTip={fakeFn}
        options={[
          {
            id: '1',
            title: 'Opção 1',
            value: 'leonardo',
          },
          {
            id: '2',
            title: 'Opção 2',
            value: 'renata',
          },
          {
            id: '3',
            title: 'Opção 3',
            value: 'robson',
          },
          {
            id: '4',
            title: 'Opção 4',
            value: 'ana-carolina',
          },
        ]}
        tooltipData={{
          title: "Título do Modal",
          description: "Descrição que aparecerá no modal, normalmente descreverá detalhes dos requisistos deste campo."
        }}
        title="Título"
      />
    </ThemeProvider>
  );
  expect(component).toMatchSnapshot()
});
