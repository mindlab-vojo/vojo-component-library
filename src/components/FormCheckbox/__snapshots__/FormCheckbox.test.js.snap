// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test FormCheckbox component render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bdfBwQ kzXhdH"
      >
        <div
          class="sc-gsTCUz fpuYvH"
        >
          <div
            class="sc-dlfnbm cxRrbB"
          >
            <div
              class="sc-hKgILt iPKFzq"
            >
              <p
                class="sc-bqyKva XcEKM"
              >
                Título
              </p>
            </div>
            <div
              class="sc-jSgupP cXdENn"
            >
              <p
                class="sc-kstrdz hEmAAc"
              >
                Este é um exemplo de descrição
              </p>
            </div>
            <div
              class="sc-jSgupP cXdENn"
            >
              <p
                class="sc-hBEYos jvBAUO"
              >
                Este é um erro
              </p>
            </div>
          </div>
          <div
            class="sc-eCssSg dTvEss"
          >
            <svg
              height="20px"
              viewBox="0 0 24 24"
              width="20px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 17h-2v-2h2v2zm2.07-7.75l-.9.92C13.45 12.9 13 13.5 13 15h-2v-.5c0-1.1.45-2.1 1.17-2.83l1.24-1.26c.37-.36.59-.86.59-1.41 0-1.1-.9-2-2-2s-2 .9-2 2H8c0-2.21 1.79-4 4-4s4 1.79 4 4c0 .88-.36 1.68-.93 2.25z"
                fill="#212529"
              />
            </svg>
          </div>
        </div>
        <div
          class="sc-gKsewC ehdtkM"
        >
          <div
            class="sc-gKsewC ehdtkM"
          >
            <div
              class="sc-iBPRYJ joNmFh"
            >
              <input
                class="sc-fubCfw ktfAJq"
                id="1"
                type="checkbox"
                value="leonardo"
              />
              <label
                class="sc-jrAGrp dkvcnq"
                for="1"
              >
                <div
                  class="sc-pFZIQ bIGLcn"
                >
                  <svg
                    height="20px"
                    viewBox="0 0 24 24"
                    width="20px"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M0 0h24v24H0V0z"
                      fill="none"
                    />
                    <path
                      d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                      fill="#B8B5C6"
                    />
                  </svg>
                </div>
                <p
                  class="sc-fodVxV irimQf"
                >
                  Opção 1
                </p>
              </label>
            </div>
            <div
              class="sc-iBPRYJ joNmFh"
            >
              <input
                class="sc-fubCfw ktfAJq"
                id="2"
                type="checkbox"
                value="renata"
              />
              <label
                class="sc-jrAGrp dkvcnq"
                for="2"
              >
                <div
                  class="sc-pFZIQ bIGLcn"
                >
                  <svg
                    height="20px"
                    viewBox="0 0 24 24"
                    width="20px"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M0 0h24v24H0V0z"
                      fill="none"
                    />
                    <path
                      d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                      fill="#B8B5C6"
                    />
                  </svg>
                </div>
                <p
                  class="sc-fFubgz kfMqys"
                >
                  Opção 2
                </p>
              </label>
            </div>
            <div
              class="sc-iBPRYJ joNmFh"
            >
              <input
                class="sc-fubCfw ktfAJq"
                id="3"
                type="checkbox"
                value="robson"
              />
              <label
                class="sc-jrAGrp dkvcnq"
                for="3"
              >
                <div
                  class="sc-pFZIQ bIGLcn"
                >
                  <svg
                    height="20px"
                    viewBox="0 0 24 24"
                    width="20px"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M0 0h24v24H0V0z"
                      fill="none"
                    />
                    <path
                      d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                      fill="#B8B5C6"
                    />
                  </svg>
                </div>
                <p
                  class="sc-bkzZxe mhzVq"
                >
                  Opção 3
                </p>
              </label>
            </div>
            <div
              class="sc-iBPRYJ joNmFh"
            >
              <input
                class="sc-fubCfw ktfAJq"
                id="4"
                type="checkbox"
                value="ana-carolina"
              />
              <label
                class="sc-jrAGrp dkvcnq"
                for="4"
              >
                <div
                  class="sc-pFZIQ bIGLcn"
                >
                  <svg
                    height="20px"
                    viewBox="0 0 24 24"
                    width="20px"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M0 0h24v24H0V0z"
                      fill="none"
                    />
                    <path
                      d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                      fill="#B8B5C6"
                    />
                  </svg>
                </div>
                <p
                  class="sc-idOhPF hPFGPq"
                >
                  Opção 4
                </p>
              </label>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bdfBwQ kzXhdH"
    >
      <div
        class="sc-gsTCUz fpuYvH"
      >
        <div
          class="sc-dlfnbm cxRrbB"
        >
          <div
            class="sc-hKgILt iPKFzq"
          >
            <p
              class="sc-bqyKva XcEKM"
            >
              Título
            </p>
          </div>
          <div
            class="sc-jSgupP cXdENn"
          >
            <p
              class="sc-kstrdz hEmAAc"
            >
              Este é um exemplo de descrição
            </p>
          </div>
          <div
            class="sc-jSgupP cXdENn"
          >
            <p
              class="sc-hBEYos jvBAUO"
            >
              Este é um erro
            </p>
          </div>
        </div>
        <div
          class="sc-eCssSg dTvEss"
        >
          <svg
            height="20px"
            viewBox="0 0 24 24"
            width="20px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 17h-2v-2h2v2zm2.07-7.75l-.9.92C13.45 12.9 13 13.5 13 15h-2v-.5c0-1.1.45-2.1 1.17-2.83l1.24-1.26c.37-.36.59-.86.59-1.41 0-1.1-.9-2-2-2s-2 .9-2 2H8c0-2.21 1.79-4 4-4s4 1.79 4 4c0 .88-.36 1.68-.93 2.25z"
              fill="#212529"
            />
          </svg>
        </div>
      </div>
      <div
        class="sc-gKsewC ehdtkM"
      >
        <div
          class="sc-gKsewC ehdtkM"
        >
          <div
            class="sc-iBPRYJ joNmFh"
          >
            <input
              class="sc-fubCfw ktfAJq"
              id="1"
              type="checkbox"
              value="leonardo"
            />
            <label
              class="sc-jrAGrp dkvcnq"
              for="1"
            >
              <div
                class="sc-pFZIQ bIGLcn"
              >
                <svg
                  height="20px"
                  viewBox="0 0 24 24"
                  width="20px"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 0h24v24H0V0z"
                    fill="none"
                  />
                  <path
                    d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                    fill="#B8B5C6"
                  />
                </svg>
              </div>
              <p
                class="sc-fodVxV irimQf"
              >
                Opção 1
              </p>
            </label>
          </div>
          <div
            class="sc-iBPRYJ joNmFh"
          >
            <input
              class="sc-fubCfw ktfAJq"
              id="2"
              type="checkbox"
              value="renata"
            />
            <label
              class="sc-jrAGrp dkvcnq"
              for="2"
            >
              <div
                class="sc-pFZIQ bIGLcn"
              >
                <svg
                  height="20px"
                  viewBox="0 0 24 24"
                  width="20px"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 0h24v24H0V0z"
                    fill="none"
                  />
                  <path
                    d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                    fill="#B8B5C6"
                  />
                </svg>
              </div>
              <p
                class="sc-fFubgz kfMqys"
              >
                Opção 2
              </p>
            </label>
          </div>
          <div
            class="sc-iBPRYJ joNmFh"
          >
            <input
              class="sc-fubCfw ktfAJq"
              id="3"
              type="checkbox"
              value="robson"
            />
            <label
              class="sc-jrAGrp dkvcnq"
              for="3"
            >
              <div
                class="sc-pFZIQ bIGLcn"
              >
                <svg
                  height="20px"
                  viewBox="0 0 24 24"
                  width="20px"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 0h24v24H0V0z"
                    fill="none"
                  />
                  <path
                    d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                    fill="#B8B5C6"
                  />
                </svg>
              </div>
              <p
                class="sc-bkzZxe mhzVq"
              >
                Opção 3
              </p>
            </label>
          </div>
          <div
            class="sc-iBPRYJ joNmFh"
          >
            <input
              class="sc-fubCfw ktfAJq"
              id="4"
              type="checkbox"
              value="ana-carolina"
            />
            <label
              class="sc-jrAGrp dkvcnq"
              for="4"
            >
              <div
                class="sc-pFZIQ bIGLcn"
              >
                <svg
                  height="20px"
                  viewBox="0 0 24 24"
                  width="20px"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 0h24v24H0V0z"
                    fill="none"
                  />
                  <path
                    d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                    fill="#B8B5C6"
                  />
                </svg>
              </div>
              <p
                class="sc-idOhPF hPFGPq"
              >
                Opção 4
              </p>
            </label>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
